### PR TITLE
warn about changing IWebHostEnvironment.WebRootPath /1

### LINKED ISSUE #25446: Add a section about the interaction during development for Static Web Assets and static files middleware
In the context of serving files from a separate folder, it is important to note that when the app uses static web assets (which is enabled by default). The assets are discovered at build time and served during development based on the locations they were discovered at.

When updating the convention to serve files from a folder different than the wwwroot folder by updating the `IWebHostEnvironment.WebRootPath` it is important to also update the static web assets configuration to reflect this fact, as otherwise, the assets inside the wwwroot folder will still be considered and served during development.

To update the configuration you can do it in one of two ways:
* Completely disable static web assets (which will prevent content from Razor Class Libraries from being taken into account) by setting up `<StaticWebAssetsEnabled>false</StaticWebAssetsEnabled>` in your project file.
* Remove the files inside the wwwroot folder from the list of implicit content items `<Content Remove="wwwroot\**" />` so that they don't get considered.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: d47f38e2-bc36-cfb8-b0cc-17a534689f8b
* Version Independent ID: 3fec6e08-fc99-7a5c-796f-3f2347cad891
* Content: [Static files in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/static-files?view=aspnetcore-6.0)
* Content Source: [aspnetcore/fundamentals/static-files.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/fundamentals/static-files.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-fundamentals**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by Rick-Anderson**
Fixes #25446
[Internal review URL](https://review.docs.microsoft.com/en-us/aspnet/core/fundamentals/static-files?view=aspnetcore-7.0&branch=pr-en-us-25471#serve-files-outside-wwwroot-by-updating-iwebhostenvironmentwebrootpath)

### CODE REVIEW

**Comment by serpent5** [comment]
@Rick-Anderson I'll review this after @javiercn.

**Review by javiercn — APPROVED** [review]

File: aspnetcore/fundamentals/static-files.md [hunk 1]
```diff
@@ -246,6 +246,42 @@ With the preceding code, a request for a file with an unknown content type is re
...
+## Serve files outside wwwroot by updating IWebHostEnvironment.WebRootPath
+
+When [IWebHostEnvironment.WebRootPat](xref:Microsoft.AspNetCore.Hosting.IWebHostEnvironment.WebRootPath) is updated to folder other than `wwwroot`:
+
+* In the development environment, static assets found in both `wwwroot` and the updated `IWebHostEnvironment.WebRootPath` are served from `wwwroot`.
+* With any environment other than development, duplicate static assets are served from the updated `IWebHostEnvironment.WebRootPath` folder.
+
+Consider a web app created with the empty web template:
+
+* Containing an *Index.html* file in *wwwroot* and *wwwroot-custom*.
```
Comment (serpent5) [inline aspnetcore/fundamentals/static-files.md:258]:
```suggestion
* Containing an `Index.html` file in `wwwroot` and `wwwroot-custom`.
```

File: aspnetcore/fundamentals/static-files.md [hunk 2]
```diff
@@ -246,6 +246,42 @@ With the preceding code, a request for a file with an unknown content type is re
...
+* With and the following updated `Program.cs` file to set `WebRootPath = "wwwroot-custom"`:
```
Comment (serpent5) [inline aspnetcore/fundamentals/static-files.md:259]:
```suggestion
* With the following updated `Program.cs` file that sets `WebRootPath = "wwwroot-custom"`:
```

File: aspnetcore/fundamentals/static-files.md [hunk 3]
```diff
@@ -246,6 +246,42 @@ With the preceding code, a request for a file with an unknown content type is re
...
+
+  [!code-csharp[](~/fundamentals/static-files/samples/6.x/WebRoot/Program.cs?name=snippet1)]
+
+In the preceding code, requests to `/`:
+
+* In the development environment return `wwwroot/Index.html`
+* In any environment other than development return `wwwroot-custom/Index.html`
+
+Any of the following approaches can be used to ensure assets from `wwwroot-custom` are returned:
```
Comment (serpent5) [inline aspnetcore/fundamentals/static-files.md:268]:
```suggestion
To ensure assets from `wwwroot-custom` are returned, use one of the following approaches:
```

File: aspnetcore/fundamentals/static-files.md [hunk 4]
```diff
@@ -246,6 +246,42 @@ With the preceding code, a request for a file with an unknown content type is re
...
+
+* Delete duplicate named assets in `wwwroot`.
+* Set `"ASPNETCORE_ENVIRONMENT"` in *Properties/launchSettings.json* to any value other than `"Development"`.
```
Comment (serpent5) [inline aspnetcore/fundamentals/static-files.md:271]:
```suggestion
* Set `"ASPNETCORE_ENVIRONMENT"` in `Properties/launchSettings.json` to any value other than `"Development"`.
```

File: aspnetcore/fundamentals/static-files.md [hunk 5]
```diff
@@ -246,6 +246,42 @@ With the preceding code, a request for a file with an unknown content type is re
...
+* Completely disable static web assets by setting `<StaticWebAssetsEnabled>false</StaticWebAssetsEnabled>` in the project file. ***WARNING, disabling static web assets disables [Razor Class Libraries](xref:razor-pages/ui-class)***.
+* Add the following JSON to the project file:
+
+  ```JSON
```
Comment (serpent5) [inline aspnetcore/fundamentals/static-files.md:275]:
```suggestion
  ```xml
```

**Review by serpent5 — APPROVED** [review]

**Pull request merged**